webhook_handlers/views/github.py

```diff
@@ -389,7 +397,44 @@ def pull_request(self, request, *args, **kwargs):
 
         return Response()
 
-    def _handle_installation_events(self, request, *args, **kwargs):
+    def _handle_installation_repository_events(self, request, *args, **kwargs):
```

Comment:
Fn looks good, but wdyt about rearranging things a bit so they're used only when they're needed? Like this (same appliies for other fns)

``` python
def _handle_installation_repository_events(self, request, *args, **kwargs):
    # https://docs.github.com/en/webhooks/webhook-events-and-payloads#installation_repositories
    service_id = request.data["installation"]["account"]["id"]
    username = request.data["installation"]["account"]["login"]
    owner, _ = Owner.objects.get_or_create(
        service=self.service_name, service_id=service_id, username=username
    )
    ghapp_installation: Optional[
        GithubAppInstallation
    ] = owner.github_app_installations.filter(
        installation_id=installation_id
    ).first()
    
    if ghapp_installation is None:
        installation_id = request.data["installation"]["id"]
        ghapp_installation = GithubAppInstallation(
            owner=owner, installation_id=installation_id
        )
    
    all_repos_affected = request.data.get("repository_selection") == "all"
    if all_repos_affected:
        ghapp_installation.repository_service_ids = None
    else:
        repo_list_to_save = set(ghapp_installation.repository_service_ids or [])
        repositories_added_service_ids = set(
            map(lambda obj: obj["id"], request.data.get("repositories_added", []))
        )
        repositories_removed_service_ids = set(
            map(lambda obj: obj["id"], request.data.get("repositories_removed", []))
        )
        repo_list_to_save = repo_list_to_save.union(
            repositories_added_service_ids
        ).difference(repositories_removed_service_ids)
        ghapp_installation.repository_service_ids = list(repo_list_to_save)
    ghapp_installation.save()
```

Reply:
Ah yes that is better indeed. Thanks for this sugestion

Reply:
np!